shapely/tests/test_io.py

```diff
@@ -579,6 +752,31 @@ def test_from_wkb_point_empty(wkb, expected_type, expected_dim):
         assert shapely.get_coordinate_dimension(geom) == expected_dim
 
 
+@pytest.mark.skipif(
+    shapely.geos_version < (3, 12, 0),
+    reason="M coordinates not supported with GEOS < 3.12",
+)
+@pytest.mark.parametrize(
+    "wkb,expected_type,expected_dim",
+    [
+        (POINTM_NAN_WKB, 0, 3),
+        (POINTZM_NAN_WKB, 0, 4),
+        (MULTIPOINTM_NAN_WKB, 4, 3),
+        (MULTIPOINTZM_NAN_WKB, 4, 4),
+        (GEOMETRYCOLLECTIONM_NAN_WKB, 7, 3),
+        (GEOMETRYCOLLECTIONZM_NAN_WKB, 7, 4),
+        (NESTED_COLLECTIONM_NAN_WKB, 7, 3),
+        (NESTED_COLLECTIONZM_NAN_WKB, 7, 4),
+    ],
+)
+def test_from_wkb_point_empty_m(wkb, expected_type, expected_dim):
+    geom = shapely.from_wkb(wkb)
+
+    assert shapely.is_empty(geom)
+    assert shapely.get_type_id(geom) == expected_type
+    assert shapely.get_coordinate_dimension(geom) == expected_dim
+
```

Comment:
Maybe add a TODO here to add an `has_m`/`has_z` check once we have those?

Reply:
Coincidentally, `has_m` will be the next PR in the series I'm working on.